platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -323,14 +352,76 @@ - (MGLImage *)blurredAttributionBackground:(CIImage *)backgroundImage
     CGImageRef cgimg = [ctx createCGImage:blurredImage fromRect:[backgroundImage extent]];
     
 #if TARGET_OS_IPHONE
-    
     return [UIImage imageWithCGImage:cgimg];
 #else
-    
     return [[NSImage alloc] initWithCGImage:cgimg size:[backgroundImage extent].size];
 #endif
 }
 
+- (MGLSnapshotAttributionOptions *)attributionOptionsForSize:(CGSize)snapshotSize attributionInfo:(NSArray *)attributionInfo
+{
+    NSMutableArray *attributionOptions = [NSMutableArray array];
+    MGLSnapshotAttributionOptions *largeLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:NO];
+#if TARGET_OS_IPHONE
+    largeLogoAttribution.logoImage = [UIImage imageNamed:@"mapbox" inBundle:[NSBundle mgl_frameworkBundle] compatibleWithTraitCollection:nil];
+#else
+    largeLogoAttribution.logoImage = [[NSImage alloc] initWithContentsOfFile:[[NSBundle mgl_frameworkBundle] pathForResource:@"mapbox" ofType:@"pdf"]];
+#endif
+    _defaultLogoHeight = largeLogoAttribution.logoImage.size.height;
+    
+    MGLSnapshotAttributionOptions *smallLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:NO];
+#if TARGET_OS_IPHONE
+    smallLogoAttribution.logoImage = [UIImage imageNamed:@"mapbox_helmet" inBundle:[NSBundle mgl_frameworkBundle] compatibleWithTraitCollection:nil];
+#else
+    smallLogoAttribution.logoImage = [[NSImage alloc] initWithContentsOfFile:[[NSBundle mgl_frameworkBundle] pathForResource:@"mapbox_helmet" ofType:@"pdf"]];
+#endif
+    
+    MGLSnapshotAttributionOptions *noLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:YES];
+    
+    [attributionOptions addObject:largeLogoAttribution];
+    [attributionOptions addObject:smallLogoAttribution];
+    [attributionOptions addObject:noLogoAttribution];
```

Comment:
Can you think of a way to defer the `logoImage` logic above until we know which size to use, so that we don’t have to put together three MGLSnapshotAttributionOptions objects?

Reply:
What do you mean? Each of these attribution options represent a rule as in https://github.com/mapbox/mapbox-gl-native/issues/10423#issue-272339917 which I use to get the best size.

Reply:
Currently, we precompute the attribution styles before selecting the most appropriate one. My suggestion was to figure out which style would be most appropriate before loading the relevant resources, like the logo, since we ultimately only care about one of the styles. That way, the `attributionOptions` array becomes unnecessary.

Reply:
We’re in a hurry to get v3.7.0 out the door, so I’ve outlined the refactor that needs to take place for v3.7.1 in #10490.